nextcord/components.py

```diff
@@ -230,43 +290,208 @@ class SelectMenu(Component):
+
+    def to_dict(self) -> RoleSelectMenuPayload:
+        payload: RoleSelectMenuPayload = {"type": self.type.value, **super().to_dict()}  # type: ignore -- fixed in pyright 1.1.278
+
+        return payload
+
+
+class MentionableSelectMenu(SelectMenuBase):
+    """Represents a mentionable select menu from the Discord Bot UI Kit.
+
+    A mentionable select menu is functionally the same as a dropdown, however
+    on mobile it renders a bit differently.
+
+    .. note::
+
+        The user constructible and usable type to create a select menu is
+        :class:`nextcord.ui.MentionableSelect` not this one.
+
+    .. versionadded:: 2.3
+
+    Attributes
+    ----------
+    custom_id: Optional[:class:`str`]
+        The ID of the select menu that gets received during an interaction.
+    placeholder: Optional[:class:`str`]
+        The placeholder text that is shown if nothing is selected, if any.
+    min_values: :class:`int`
+        The minimum number of items that must be chosen for this select menu.
+        Defaults to 1 and must be between 1 and 25.
+    max_values: :class:`int`
+        The maximum number of items that must be chosen for this select menu.
+        Defaults to 1 and must be between 1 and 25.
+    disabled: :class:`bool`
+        Whether the select is disabled or not. Defaults to ``False``.
+    """
+
+    __slots__: Tuple[str, ...] = ()
+
+    __repr_info__: ClassVar[Tuple[str, ...]] = __slots__
+
+    def __init__(self, data: MentionableSelectMenuPayload):
+        super().__init__(data)
+        self.type = ComponentType.mentionable_select
+
+    def to_dict(self) -> MentionableSelectMenuPayload:
+        payload: MentionableSelectMenuPayload = {"type": self.type.value, **super().to_dict()}  # type: ignore -- fixed in pyright 1.1.278
+
+        return payload
+
+
+class ChannelSelectMenu(SelectMenuBase):
+    """Represents a mentionable select menu from the Discord Bot UI Kit.
+
+    A mentionable select menu is functionally the same as a dropdown, however
+    on mobile it renders a bit differently.
+
+    .. note::
+
+        The user constructible and usable type to create a select menu is
+        :class:`nextcord.ui.ChannelSelect` not this one.
+
+    .. versionadded:: 2.3
+
+    Attributes
+    ----------
+    custom_id: Optional[:class:`str`]
+        The ID of the select menu that gets received during an interaction.
+    placeholder: Optional[:class:`str`]
+        The placeholder text that is shown if nothing is selected, if any.
+    min_values: :class:`int`
+        The minimum number of items that must be chosen for this select menu.
+        Defaults to 1 and must be between 1 and 25.
+    max_values: :class:`int`
+        The maximum number of items that must be chosen for this select menu.
+        Defaults to 1 and must be between 1 and 25.
+    disabled: :class:`bool`
+        Whether the select is disabled or not. Defaults to ``False``.
+    channel_types: List[:class:`nextcord.ChannelType`]
```

Comment:
Could these `nextcord.*` references use just the member name - e.g. `ChannelType`, `SelectOption`, so it doesn't look so big.
![image](https://user-images.githubusercontent.com/34910574/200667265-6a315f0e-7fa1-4030-9aa9-b02401635a89.png)